### PR TITLE
Add rule controller

### LINKED ISSUE #90: Create controller for adding rules

### PR DESCRIPTION
**Pull request by thu5**
This PR adds the controller to handle operations on rules. Resolves #90.

### CODE REVIEW

File: server/test/rule_controller_test.dart [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+          MockHttpRequest('POST', Uri.parse('/rules'))
+            ..headers.add('Content-Type', 'application/x-protobuf')
+            ..headers.add('Authorization', 'Bearer $idToken'));
+      response =
+          await ruleController.createRule(createRuleRequest.writeToBuffer());
+      ruleWithId = rule.clone()..id = ruleId;
+    });
+
+    test('calls FirestoreClient.createRule() with user ID', () async {
+      verify(mockFirestoreClient.createRule(userId, any));
```
Comment (cathyang-work) [inline server/test/rule_controller_test.dart]: can join the 2 tests. `verify(mockFirestoreClient.createRule(userId, ruleWithId));` Ditto for ` verify(mockFirestoreClient.createRule(userId, any));`

File: server/lib/controller/rule_controller.dart [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  @Operation.post()
+  Future<Response> createRule(@Bind.body() List<int> body) async {
+    // The ID token is sent in the `Authorization` header of the request.
+    final authorizationHeader = request.raw.headers.value('Authorization');
+    if (authorizationHeader == null) {
+      throw ArgumentError('Request does not contain an Authorization header');
+    }
+    final idToken = authorizationHeader.split(' ').last;
+
+    final userId = getUserId(idToken);
```
Comment (gg-qinp) [inline server/lib/controller/rule_controller.dart:42]: this is used in multiple places, maybe consider extract them into util class?

File: server/test/rule_controller_test.dart (hunk 1 above)
Comment (thu5) [reply server/test/rule_controller_test.dart]: Okay!

File: server/lib/controller/rule_controller.dart (hunk 2 above)
Comment (thu5) [reply server/lib/controller/rule_controller.dart:42]: Ok, I've created an [issue](https://github.com/googleinterns/dv360-excel-plugin/issues/112), and I'll resolve it after all these controllers are merged in.

**Review by cathyang-work — APPROVED** [review]

**Review by gg-qinp — APPROVED** [review]

**Pull request merged**